File: weekOfOct2/isMonotonic.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isMonotonic = function(nums) {
  if (nums.length === 1) return true;

  let increasing = nums[0] === nums[1] ?
      undefined :
      nums[0] > nums[1] ?
          false : true;

  for (let i = 2; i < nums.length; i++) {
      if (increasing === undefined) increasing = nums[i - 1] === nums[i] ?
                                                      undefined :
                                                      nums[i - 1] > nums[i] ?
                                                          false : true;

      else if (increasing && nums[i - 1] > nums[i]) return false
      else if (!increasing && nums[i - 1] < nums[i]) return false
  }

  return true
};

File: weekOfOct2/isMonotonic.py
class Solution(object):
    def isMonotonic(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """


        if len(nums) == 1:
            return True

        increasing = None

        for i in range (1, len(nums)):
            print(increasing)
            if increasing == None:
                increasing = None if nums[i - 1] == nums[i] else False if nums[i - 1] > nums[i] else True
            elif increasing and nums[i - 1] > nums[i]:
                return False
            elif not increasing and nums[i - 1] < nums[i]:
                return False

        return True
